structuralcodes/sections/section_integrators/_marin_integrator.py

```diff
@@ -176,47 +245,143 @@ def integrate(
 
+    def integrate_modulus(
+        self,
+        angle: float,
+        prepared_input: t.List[
+            t.Tuple[int, np.ndarray, np.ndarray, np.ndarray]
+        ],
+    ) -> NDArray[np.float64]:
+        """Integrate material modulus over the geometry.
+
+        Arguments:
+            prepared_input (List): The prepared input from .prepare_input().
+
+        Returns:
+            ndarray: The section stiffness matrix as a (3, 3) ndarray.
+        """
+        # Create the stiffness matrix
+        stiffness = np.zeros((3, 3), dtype=np.float64)
+
+        # Create a map of indices
+        # (i,j), m, offset, sign
+        indices = [
+            ((0, 0), 0, 0, 1),
+            ((0, 1), 0, 1, 1),
+            ((0, 2), 1, 0, -1),
+            ((1, 1), 0, 2, 1),
+            ((1, 2), 1, 1, -1),
+            ((2, 2), 2, 0, 1),
+        ]
+
+        # Loop through all parts of the section and add contributions
+        for id, y, z, modulus_coeff in prepared_input:
+            if id == 0:
+                # Find integration order from shape of stress coeff array
+                n = modulus_coeff.shape[0]
+
+                # Calculate needed area moments
+                for (i, j), m, offset, sign in indices:
+                    area_moments = np.array(
+                        [
+                            marin_integration(y, z, m, k + offset)
+                            for k in range(n)
+                        ]
+                    )
+                    stiffness[i, j] += sign * sum(modulus_coeff * area_moments)
+            elif id == 1:
+                # Reinforcement
+                stiffness[0, 0] += sum(modulus_coeff)
+                stiffness[0, 1] += sum(modulus_coeff * z)
+                stiffness[0, 2] -= sum(modulus_coeff * y)
+                stiffness[1, 1] += sum(modulus_coeff * z * z)
+                stiffness[1, 2] -= sum(modulus_coeff * y * z)
+                stiffness[2, 2] += sum(modulus_coeff * y * y)
+
+        # Apply for simmetry
+        stiffness[1, 0] = stiffness[0, 1]
+        stiffness[2, 0] = stiffness[0, 2]
+        stiffness[2, 1] = stiffness[1, 2]
+
+        # Rotate back to section CRS
+        T = np.array([[cos(-angle), -sin(-angle)], [sin(-angle), cos(-angle)]])
+        T = T @ np.array([[100], [-100]])
+
+        T = np.array(
+            [
+                [1, 0, 0],
+                [0, cos(-angle), sin(-angle)],
+                [0, -sin(-angle), cos(-angle)],
+            ]
+        )
+
+        return T.T @ stiffness @ T
+
     def integrate_strain_response_on_geometry(
-        self, geo: CompoundGeometry, strain: ArrayLike, **kwargs
+        self,
+        geo: CompoundGeometry,
+        strain: ArrayLike,
+        integrate: t.Literal['stress', 'modulus'] = 'stress',
+        **kwargs,
     ):
-        """Integrate the strain response with the Marin algorithm.
+        """Integrate the strees from strain response with the Marin algorithm.
```

Comment:
Suggest to modify the docstring similar to the suggestions above.

Reply:
done